Riot/Managers/UISIAutoReporter/UISIAutoReporter.swift

```diff
@@ -0,0 +1,222 @@
+            eventId: source.eventId,
+            roomId: source.roomId,
+            senderKey: source.senderKey,
+            deviceId: source.senderDeviceId,
+            userId: source.senderUserId,
+            sessionId: source.sessionId
+        ).jsonString ?? ""
+        
+        self.bugReporter.vc_sendBugReport(
+            description: "Auto-reporting decryption error",
+            sendLogs: true,
+            sendCrashLog: true,
+            additionalLabels: [
+                "Z-UISI",
+                "ios",
+                "uisi-recipient"
+            ],
+            customFields: ["auto_uisi": uisiData],
+            success: { reportUrl in
+                let contentMap = MXUsersDevicesMap<NSDictionary>()
+                let content = [
+                    "event_id": source.eventId,
+                    "room_id": source.roomId,
+                    "session_id": source.sessionId,
+                    "device_id": source.senderDeviceId,
+                    "user_id": source.senderUserId,
+                    "sender_key": source.senderKey,
+                    "recipient_rageshake": reportUrl
+                ]
+                contentMap.setObject(content as NSDictionary, forUser: source.senderUserId, andDevice: source.senderDeviceId)
+                session.matrixRestClient.sendDirectToDevice(
+                    eventType: Self.autoRsRequest,
+                    contentMap: contentMap,
+                    txnId: nil
+                ) { response in
+                    if response.isFailure {
+                        MXLog.warning("failed to send auto-uisi to device")
+                    }
+                }
+            },
+            failure: { [weak self] error in
+                guard let self = self else { return }
+                self.dispatchQueue.async {
+                    self.alreadyReportedUisi.remove(ReportInfo(roomId: source.roomId, sessionId: source.sessionId))
+                }
+            })
+    }
+    
+    func sendMatchingRageShake(source: MXEvent) {
+        MXLog.debug("[UISIAutoReporter] sendMatchingRageShake")
+        let eventId = source.content["event_id"] as? String
+        let roomId = source.content["room_id"] as? String
+        let sessionId = source.content["session_id"] as? String
+        let deviceId = source.content["device_id"] as? String
+        let userId = source.content["user_id"] as? String
+        let senderKey = source.content["sender_key"] as? String
+        let matchingIssue = source.content["recipient_rageshake"] as? String ?? ""
+        
+        
+        let uisiData = UISIAutoReportData(
+            eventId: eventId,
+            roomId: roomId,
+            senderKey: senderKey,
+            deviceId: deviceId,
+            userId: userId,
+            sessionId: sessionId
+        ).jsonString ?? ""
+        
+        self.bugReporter.vc_sendBugReport(
+            description: "Auto-reporting decryption error",
```

Comment:
One thing that'd be good to add here: if the `description` field contains the `recipient_rageshake` URL on a second line, it'll show up linkified in the github issue body so it's a lot easier to cross-reference sender and recipient rageshakes. See https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/stores/AutoRageshakeStore.ts#L149